### PR TITLE
Only running bouts place cells

### PR DESCRIPTION
**Pull request by neuromantic99**
## Summary by CodeRabbit

- New Features
  - Added Grosmark preprocessing pipeline with denoising, deconvolution, and spike outputs.
  - Introduced spike binarisation utility and new imaging helpers for speed and online position.
  - Expanded genotype/sex mappings; added threshold-run detection utility.

- Refactor
  - Streamlined place-field workflow; externalized spike binarisation.
  - Deprecated a legacy speed function in favor of a new approach.
  - Tightened anticipatory-lick timing window.

- Documentation
  - Removed OASIS submodule installation section from README.

- Tests
  - Added coverage for new utilities and edge cases.

- Chores
  - Updated and trimmed dependencies; minor ignore rule reordering.

### CODE REVIEW

File: viral/imaging_utils.py [hunk 1]
```diff
@@ -112,6 +115,100 @@ def extract_TTL_chunks(
...
+    return speed
+
+
+def get_online_position_and_frames(
+    trial: TrialInfo, wheel_circumference: float
+) -> Tuple[np.ndarray, np.ndarray]:
+    """Offline immobility epochs were defined as those in which the animal's velocity,
+    smoothed with a half-second Gaussian kernel, was below 3cms-1 for at least 3 consecutive seconds.
+    Online running epochs were defined as those in which the animal's smoothed velocity was above 5cms-1
+    for at least 3 consecutive seconds."""
```
Comment (josefbitzenhofer) [inline viral/imaging_utils.py:176]: Weird definition of immobility but we talked about it, I think? You took that from Grosmark, right?

File: viral/imaging_utils.py [hunk 2]
```diff
@@ -112,6 +115,100 @@ def extract_TTL_chunks(
...
+
+    speed_threshold = 5
+    idx_keep = above_threshold_for_n_consecutive_samples(
+        speed, threshold=speed_threshold, n_samples=3 * 30
+    )
+
+    # Removed the first two seconds as there is a bit of a burst of activity when the screens come on, which is not unexpected
+    trial_onset = frame_position < frame_position[0] + 60
+    # TODO: Make sure this works
+    idx_keep = idx_keep & ~trial_onset
```
Comment (josefbitzenhofer) [inline viral/imaging_utils.py:202]: Aaaaand? Does it?

File: viral/imaging_utils.py [hunk 3]
```diff
@@ -166,6 +249,8 @@ def activity_trial_position(
             print(f"bin_start: {bin_start}")
             print(f"bin_end: {bin_start + bin_size}")
             print(f"n_frames in bin: {len(frame_idx_bin)}")
+
+        # Does this answer David's question? We're averaging over the frames in the bin. So does this constitute controlling for speed?
```
Comment (josefbitzenhofer) [inline viral/imaging_utils.py:267]: If I understand correctly, a bin contains all frames within a range of positions, e.g. from 170 - 180 cm, and the activity is averaged over these frames. I think it does account for the animal spending more time "in a bin" because its speed was lower. Don't know if from the neuroscience point of view there still might be some biased activity, e.g. if the mouse is then more likely to show whisking although that shouldn't happen to much when the mouse is still running a bit - and you are filtering out the only running bouts above. I'd be curious to hear thoughts and/or further discussion.

File: viral/ensemble_reactivation.py [hunk 4]
```diff
@@ -0,0 +1,866 @@
...
+            start=0,
+            end=170,
+        )
+        spks_raw, _, _, _ = load_data(
+            session,
+            # AHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHH
+            Path("/Volumes/hard_drive/VR-2p/2025-03-25/JB031/suite2p/plane0"),
+            # TIFF_UMBRELLA / session.date / session.mouse_name / "suite2p" / "plane0",
+            "spks",
+        )
```
[the remark this answers is not in the document]
Comment (josefbitzenhofer) [reply viral/ensemble_reactivation.py:635]: Do we care about this now? Or will this be superceded by another PR?
